### DISCUSSION TITLE
3.2.2 二进制包内缺少部署集群的脚本 且缺少install脚本

### DISCUSSION THREAD

**Discussion by mengbaba3316** [apache/dolphinscheduler#16357] [Q&A]
版本：3.2.2
问题：3.2.2 二进制包内，没有部署cluster配置脚本。官网文档缺少集群部署相关说明。
我查看了一下 3.2.1 一般集群化配置是在 bin/env/下的  install_env.sh 中进行配置，到那时在 3.2.2包内 的 该路径下并没有 这个文件，而且 之前的版本在  /bin下还会有一个 install.sh  这个脚本在 初始化 数据库之后 执行的，但是 在 3.2.2包内 也并没有 发现。
大家可以查看一下下面的图片
![6dccad58898b606fc3a24525a65f90b](https://github.com/user-attachments/assets/b5d37c24-324c-4cf0-9ea1-37679aba7a3f)
![1721628236679](https://github.com/user-attachments/assets/075a533a-499b-4ae6-a5aa-66ad9c3b78c0)

**Comment by Fortytwoo** [comment]
我也遇到了这个问题，解决办法是暂时使用3.1.9版本。这是最新具有install脚本的版本。
我认为dolphinscheduler的文档很糟糕，部署以及搭建中有着大量的不可预知的行为。他还需要完善。

**Reply by mengbaba3316** [reply]
是的 但是这个问题我好像解决了，对于install_env.sh。我对比了之前版本的脚本之后，基于3.2.2版本的海豚，install_env这个脚本是必须的，但是dolphineche_home/bin 下面的install 是不必须的，首先 你可以不接受官网的建议 然后  直接copy 一份之前版本的  installenv 和install 脚本到对应的环境目录下就能执行。installenv 里面有 集群的配置信息，而且startall 和 stopall在启动集群的时候 都要去 install env去获取集群的安装信息，故 install env 必然需要，install 是纯粹分发软件包的 所以 不是必须的 这也就对应了 上图 官方说的  scp 自己去分发的意思了。这样做确实有点麻烦。反正整体，来说 3.2.2  是能用的。你可以回头再尝试一下

**Comment by liuzhao900928** [comment]
大佬有没有遇到过3.2.2部署3台服务器的集群模式时候，集群启动后master-server、api-server、alert-server 很快down掉的情况，worker-server没问题，看报错日志基本都是缺少相关类，如果遇到的话帮忙看下这个问题
![dolphinschedular1](https://github.com/user-attachments/assets/f467d408-a97b-4e02-95c1-a4ab4e24c67d)
![Uploading dolphinschedular2.jpg…]()

**Reply by ElyCman** [reply]
3.2.2版本二进制包，解压后每个libs目录都缺少mysql-connector-j-xx.jar包。这个需要自行下载。去3.1.9版本下载install_env.sh和install.sh 这两个文件，放到对应目录下。安装后就能正常启动

**Reply by liuzhao900928** [reply]
mysql-connector-j-xx.jar这个包我已经上传至每个内部组件的libs目录下，install_env.sh也已从旧版本下载上传至/apache-dolphinschedular-3.2.2-bin/bin/env下，但是还是有同样的问题。这个install.sh脚本我在3.1.9和3.2.2中都没找到

**Reply by ElyCman** [reply]
直接在这上边3.1.9版本，scrip目录下——我记得是这个目录。你找找

**Reply by ElyCman** [reply]
<img width="642" height="1389" alt="IMG_9312" src="https://github.com/user-attachments/assets/a14a17d2-96f5-47fe-8f1a-b75980bbdebb" />

**Reply by liuzhao900928** [reply]
install.sh我在3.1.9版本海豚中也找到了，但是我看楼上说这个主要是做分发用的，主要还是install_env.sh这个文件使用，mysql的connector包我也上传到每个环节的libs目录下了，但是重新启动还是有我上面图片的问题
